pkg/envoy/rds.go

```diff
@@ -46,69 +47,83 @@ func (l RouteDomainCheck) Run() outcomes.Outcome {
 			return outcomes.Fail{Error: ErrUnmarshalingDynamicRouteConfig}
 		}
 
-		if dynRouteCfg.Name != l.RouteName {
+		if dynRouteCfg.Name != check.RouteName {
 			continue
 		}
 
 		for _, virtualHost := range dynRouteCfg.GetVirtualHosts() {
 			for _, domain := range virtualHost.GetDomains() {
 				foundAnyRouteDomains = true
-				if l.Domain == "" {
+				if len(check.Domains) == 0 {
 					break
 				}
-				if domain == l.Domain {
+				if _, ok := check.Domains[domain]; ok {
 					foundSpecificRouteDomain = true
 					break
 				}
 			}
-			if (l.Domain == "" && foundAnyRouteDomains) || foundSpecificRouteDomain {
+			if (len(check.Domains) == 0 && foundAnyRouteDomains) || foundSpecificRouteDomain {
```

Comment:
nit: In the OSM repo we primarily compare strings with an empty string. For consistency, we should continue to use the empty string comparison.